iiod/iiod.c

```diff
@@ -407,9 +407,24 @@ static int main_interactive(struct iio_context *ctx, bool verbose, bool use_aio)
 
 	if (!use_aio) {
 		flags = fcntl(STDIN_FILENO, F_GETFL);
-		fcntl(STDIN_FILENO, F_SETFL, flags | O_NONBLOCK);
+		if (flags > 0)
+			flags = fcntl(STDIN_FILENO, F_SETFL, flags | O_NONBLOCK);
+		if (flags < 0) {
+			char err_str[1024];
+			iio_strerror(errno, err_str, sizeof(err_str));
+			ERROR("Could not get/set O_NONBLOCK on STDIN_FILENO"
+					" %s (%d)\n", err_str, -errno);
+		}
+
 		flags = fcntl(STDOUT_FILENO, F_GETFL);
-		fcntl(STDOUT_FILENO, F_SETFL, flags | O_NONBLOCK);
+		if (flags > 0)
+			flags = fcntl(STDOUT_FILENO, F_SETFL, flags | O_NONBLOCK);
```

Comment:
same question here about `flags >= 0`
thing is: depending on the case, `flags >= 0` can change behavior or introduce a new bug

Reply:
fixed / Thanks